### PR TITLE
[KernelGen][Nvidia] Add _batch_norm_with_update_functional operator with Triton Kernel

### PR DESCRIPTION
**Pull request by YufanPeter**
### Summary

Adds a implementation of `_batch_norm_with_update_functional` (`torch.ops.aten._batch_norm_with_update_functional`), the functional variant of training-mode batch normalization.

The forward kernel reuses the proven attorch-derived `batch_norm_forward_kernel` (the same kernel backing `flag_gems.batch_norm` and `_batch_norm_with_update`): a 1D grid over channels, each program running a serial Welford reduction over its `(batch × spatial)` slice, followed by a normalization pass in the same launch. Implementation details:

- Returns the 6-tuple `(output, save_mean, save_invstd, reserve, running_mean_out, running_var_out)` matching the aten signature.
- Uses separate input/output running-stat buffers: the kernel reads the old `running_mean` / `running_var` and writes updated values into new tensors, preserving functional (non-mutating) semantics.
- `save_invstd` returns `inv_std = 1/sqrt(var + eps)`, following PyTorch's convention.
- `reserve` is an empty `uint8` tensor, consistent with the Triton fallback path (neither cuDNN nor MIOpen).

### Testing

- Parametrized over 3D / 4D / 5D input shapes, affine and non-affine modes, and dtypes `float16 / float32 / bfloat16`
- Coverage: normalized output correctness, `save_mean` / `save_invstd` values, and returned running-statistics updates
- **All 30 accuracy cases pass**

### Performance

Test command: `pytest benchmark/test__batch_norm_with_update_functional.py --level core` (NVIDIA H20-3e, warm cache)

> Baseline is `torch.ops.aten._batch_norm_with_update_functional` (same 6-tuple interface), so the comparison is apples-to-apples. Throughput scales with channel count, since the kernel parallelizes over channels only.

| dtype | shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------|--------------------|-------------------|---------|
| float16 | [4, 16, 64, 4] | 0.016416 | 0.008736 | 1.879 |
| float16 | [16, 16, 8, 48] | 0.022336 | 0.009360 | 2.386 |
| float16 | [16, 16, 8, 88] | 0.028544 | 0.010592 | 2.695 |
| float16 | [16, 16, 128] | 0.016448 | 0.007488 | 2.197 |
| float16 | [20, 6, 65536] | 1.073984 | 0.334592 | 3.210 |
| float32 | [4, 16, 64, 4] | 0.010496 | 0.008128 | 1.291 |
| float32 | [16, 16, 8, 48] | 0.012544 | 0.009120 | 1.375 |
| float32 | [16, 16, 8, 88] | 0.014976 | 0.011200 | 1.337 |
| float32 | [16, 16, 128] | 0.010976 | 0.007008 | 1.566 |
| float32 | [20, 6, 65536] | 0.594192 | 0.528560 | 1.124 |
| bfloat16 | [4, 16, 64, 4] | 0.015904 | 0.009216 | 1.726 |
| bfloat16 | [16, 16, 8, 48] | 0.022432 | 0.008672 | 2.587 |
| bfloat16 | [16, 16, 8, 88] | 0.028544 | 0.010784 | 2.647 |
| bfloat16 | [16, 16, 128] | 0.016416 | 0.007520 | 2.183 |
| bfloat16 | [20, 6, 65536] | 1.074160 | 0.339584 | 3.163 |

#### Arithmetic Mean Speedup

| dtype | Mean Speedup |
|-------|--------------|
| float16 | 2.473 |
| float32 | 1.339 |
| bfloat16 | 2.461 |
| **Overall** | **2.091x** |

### Multi-backend Testing

| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20-3e) | PASS (30 cases) | 2.091 (overall) | Primary (fp16 / fp32 / bf16); channel-parallel kernel, best on large tensors |

### Files Changed

- `src/flag_gems/ops/_batch_norm_with_update_functional.py`: Triton implementation (attorch-derived batch_norm forward kernel, functional semantics)
- `tests/test_batch_norm_with_update_functional.py`: Accuracy test
- `benchmark/test__batch_norm_with_update_functional.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to config
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -13,6 +13,17 @@
 # limitations under the License.
 
 ops:
+  - id: _batch_norm_with_update_functional
```
Comment (Caeruleann) [inline conf/operators.yaml]:
```suggestion
  - id: batch_norm_with_update_functional
```

File: benchmark/test__batch_norm_with_update_functional.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    weight = torch.randn((C,), dtype=dtype, device=device)
+    bias = torch.randn((C,), dtype=dtype, device=device)
+    running_mean = torch.randn((C,), dtype=dtype, device=device)
+    running_var = torch.abs(torch.randn((C,), dtype=dtype, device=device)) + 0.1
+    momentum = 0.1
+    eps = 1e-5
+    yield inp, weight, bias, running_mean, running_var, momentum, eps
+
+
+@pytest.mark._batch_norm_with_update_functional
```
Comment (Caeruleann) [inline benchmark/test__batch_norm_with_update_functional.py]:
```suggestion
@pytest.mark.batch_norm_with_update_functional
```
